crates/bevy_ecs/src/world/unsafe_world_cell.rs

```diff
@@ -709,6 +729,25 @@ impl<'w> UnsafeEntityCell<'w> {
             world,
             entity,
             location,
+            last_run: world.last_change_tick(),
+            this_run: world.change_tick(),
+        }
+    }
+
+    #[inline]
+    pub(crate) fn new_with_ticks(
```

Comment:
Instead of adding a separate `new` method, it might make sense to add the parameters to the existing one.  That will make it harder to forget to provide ticks for callers that have them.  

It looks like the only remaining calls that aren't passing ticks are some conversion methods in `EntityWorldMut`, though.  That will be using the right value anyway, since `world.change_tick` is updated during exclusive systems.  (Although it's *slightly* more efficient to read it using `World::change_tick` instead of `UnsafeWorldCell::change_tick` since it can do a non-atomic load.)

Reply:
Yes I prefer that; I disliked having an additional `new` method